pkg/translator/prometheusremotewrite/helper.go

```diff
@@ -164,12 +164,15 @@ func createAttributes(resource pcommon.Resource, attributes pcommon.Map, externa
 
 	// Ensure attributes are sorted by key for consistent merging of keys which
 	// collide when sanitized.
-	// Sorting is done on a cloned map, as the original attributes map can read at
-	// the same time in different places.
-	cloneAttributes := pcommon.NewMap()
-	attributes.CopyTo(cloneAttributes)
-	cloneAttributes.Sort()
-	cloneAttributes.Range(func(key string, value pcommon.Value) bool {
+	sortedKeys := make([]string, 0, attributes.Len())
+	attributes.Range(func(key string, value pcommon.Value) bool {
+		sortedKeys = append(sortedKeys, key)
+		return true
+	})
+	sort.Strings(sortedKeys)
+
+	for _, key := range sortedKeys {
+		value, _ := attributes.Get(key)
```

Comment:
Is it possible for a map to have multiple entries with the same key?  I believe the answer is yes.  What effect does this change have on the correctness of this translation in such a case?

Reply:
It is possible but it'd invalid data. This commit doesn't change the behaviour in such cases

Reply:
Actually I'm not 100% sure. It might be needed to apply stable sort to preserve exactly the same behaviour for such cases, I'll double check it in an hour

Reply:
Yes, it needs stable sort to preserve the same behavior. Correctness is good in any case. It looks like benchmark results are pretty much the same, so I applied the stable sort instead.

Reply:
Ok, it looks like the order of the keys doesn't matter. This approach will always introduce duplicate key/value pairs if the map has duplicate keys. We can make sure that `sortedKeys` doesn't have any duplicates, at least to mitigate this corner case. But given that it's invalid data with undefined behavior, I'm not sure we should handle it at al.

Anyway, I made the changes to replicate the existing behavior. It complicates the code and shows not as good benchmark results:

```
BenchmarkCreateAttributes
BenchmarkCreateAttributes-10    	 1545469	       772.1 ns/op	     627 B/op	       8 allocs/op
```

Again, I would rather revert this to the original solution, but let me know what you think.

Reply:
> Is it possible for a map to have multiple entries with the same key? I believe the answer is yes.

It's true for now, but we might change pcommon.Map in the future to hide/remove the duplicate key from the user, even with Map.Range.

Reply:
@Aneurysm9, please take a look

Reply:
I think I'd prefer to have this change as it now stands without behavior change.  It still captures the allocation savings and half of the runtime savings.  The behavior in the presence of overloaded keys can and should be addressed separately from this refactoring.